docs/sorting-algorithms/insertion-sort.mdx

```diff
@@ -0,0 +1,194 @@
+---
+title: Insertion Sort
+slug: /sorting-algorithm/insertion-sort
+---
+
+## Introduction
+
+A simple sorting algorithm which works on the idea of picking elements from the unsorted part one by one and inserting them into their right positions in the sorted part. The array is virtually divided into 2 parts, a sorted one and an unsorted one. Just like we arrange a group of shuffled pages according to their page numbers, values form unsorted part are picked and placed at the right positions in the sorted portion.
+
+![Imgur](https://i.imgur.com/Ora9Q1J.png)
+
+## Algorithm
+
+For sorting an array of size n in ascending order:
+1. Start your loop from arr[1] till arr[n].
+2. Compare the selected element of the current iteration (key) to the element before it.
+3. If this key is smaller than its predecessor, compare it to the elements before the predecessor.
+4. Shift all the greater elements up by one position to make space for the smaller element.
+
+## Code
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+<Tabs
+  defaultValue="cpp"
+  values={[
+    { label: 'C++', value: 'cpp', },
+    { label: 'Python', value: 'py', },
+  ]
+}>
+
+<TabItem value="cpp">
+
+```cpp
+// EXECUTION OF INSERTION SORT IN C++
+
+/*First we  will  compare first element in the list with its adjacent element.
+And at every comparison if the element can be inserted at a particular position, 
+then it is inserted by shifting the other elements one position to the right.
+The above steps are repeated until all the elements in the list are in their 
+appropriate position.i.e. list is in sorted order.*/
+
+
+
+
+#include <iostream>
+using namespace std;
+
+//Function to print array.
+void display(int arr[], int size) 
+{ 
+    int i; 
+    for (i=0; i < size; i++) 
+    cout<< arr[i]<<"\t"; 
+    cout<<"\n"; 
+} 
+
+//Main function to run the program.
+int main() 
+{ 
+    int array[] = {5, 3, 1, 9, 8, 2, 4,7}; 
+    int size = sizeof(array)/sizeof(array[0]); 
+
+    cout<<"Before Insertion sort: \n";
+    display(array, size);//Using dispaly function to print unsorted array.
+
+    int i, target, j;  
+    for (i = 1; i < size; i++) 
+    {  
+        target = array[i];  
+        j = i - 1;  
+
+        /* Here the elements in b/w arrary[0 to i-1]
+        which are greater than target are moved
+        ahead by 1 position each*/
+        while (j >= 0 && array[j] > target) 
+        {  
+            array[j + 1] = array[j];  
+            j = j - 1;  
+        }  
+        array[j + 1] = target;  
+    }
+    cout<<"After Insertion sort: \n"; 
+    display(array, size);//Using dispaly function to print sorted array. 
+    return 0; 
+}
```

Comment:
Can you check the C++ code for plagiarism? It seems kind of suspicious.

Reply:
@HarshCasper I compared this with code on GFG, and I saw that the main steps of the algorithm are the same except for different variable names and different documentation. Would this be counted in Plagiarism? And how should we resolve it?

Reply:
@HarshCasper I compared this with code on GFG, and I saw that the main steps of the algorithm are the same except for different variable names and different documentation. Would this be counted in Plagiarism? And how should we resolve it?

Reply:
Yes. It would be counted as plagiarism. Kindly fix the implementation by writing your own code and send a PR over to NeoAlgo as well for the same. We are sorry for the extra maintenance overload but we are trying to make sure that the maintenance for docs is unhindered.

Reply:
@ritvij14 Rajiv is not responsible for the plagiarism. I traced the PR here: https://github.com/TesseractCoding/NeoAlgo/pull/52/files 

Ricardo took it over and merged it up. So I guess you can write your own implementation and credit yourself for the same and send a PR to NeoAlgo to fix it up.

Reply:
> Ricardo took it over and merged it up. So I guess you can write your own implementation and credit yourself for the same and send a PR to NeoAlgo to fix it up.

Alright got it, I will send a PR for that. But in NeoAlgo, 2 of my issues are already open, so should I close one of them and then open one for this or should I do this after completing those 2?

Reply:
Just send a PR directly. No need of an Issue for that. Code refactors, bug fixes and fixing dead code requires no issues.

Reply:
@HarshCasper done, please have a look -> https://github.com/TesseractCoding/NeoAlgo/pull/5472